### ISSUE TITLE
Scroll not working for ReactiveDatePicker focus

### ISSUE THREAD

**Issue by mammba01012018** [joanpablo/reactive_forms#137]
Here's the sample code. Just edited the example project included

`import 'package:flutter/material.dart' hide ProgressIndicator;
import 'package:reactive_forms/reactive_forms.dart';
import 'package:reactive_forms_example/progress_indicator.dart';
import 'package:reactive_forms_example/sample_screen.dart';

class ComplexSample extends StatelessWidget {
  FormGroup buildForm() => fb.group(<String, dynamic>{
        'email': FormControl<String>(
          validators: [Validators.required, Validators.email],
          asyncValidators: [_uniqueEmail],
        ),
        'password': ['', Validators.required, Validators.minLength(8)],
        'passwordConfirmation': '',
        'dateTime': DateTime.now(),
      }, [
        Validators.mustMatch('password', 'passwordConfirmation')
      ]);

  @override
  Widget build(BuildContext context) {
    return SampleScreen(
      title: const Text('Complex sample'),
      body: ReactiveFormBuilder(
        form: buildForm,
        builder: (context, form, child) {
          return Column(
            children: [
              ReactiveTextField<DateTime>(
                formControlName: 'dateTime',
                readOnly: true,
                decoration: InputDecoration(
                  labelText: 'Birthday',
                  suffixIcon: ReactiveDatePicker<DateTime>(
                    formControlName: 'dateTime',
                    firstDate: DateTime(1985),
                    lastDate: DateTime(2030),
                    builder: (context, picker, child) {
                      return IconButton(
                        onPressed: picker.showPicker,
                        icon: const Icon(Icons.date_range),
                      );
                    },
                  ),
                ),
              ),
              const SizedBox(height: 20.0),
              ReactiveTextField<String>(
                formControlName: 'email',
                validationMessages: (control) => {
                  ValidationMessage.required: 'The email must not be empty',
                  ValidationMessage.email:
                      'The email value must be a valid email',
                  'unique': 'This email is already in use',
                },
                onSubmitted: () => form.focus('password'),
                textInputAction: TextInputAction.next,
                decoration: InputDecoration(
                  labelText: 'Email',
                  suffixIcon: ReactiveStatusListenableBuilder(
                    formControlName: 'email',
                    builder: (context, control, child) => Visibility(
                      visible: control.pending,
                      child: ProgressIndicator(),
                    ),
                  ),
                ),
              ),
              const SizedBox(height: 400.0),
              ReactiveTextField<String>(
                formControlName: 'password',
                obscureText: true,
                validationMessages: (control) => {
                  ValidationMessage.required: 'The password must not be empty',
                  ValidationMessage.minLength:
                      'The password must be at least 8 characters',
                },
                onSubmitted: () => form.focus('passwordConfirmation'),
                textInputAction: TextInputAction.next,
                decoration: const InputDecoration(labelText: 'Password'),
              ),
              const SizedBox(height: 24.0),
              ReactiveTextField<String>(
                formControlName: 'passwordConfirmation',
                decoration:
                    const InputDecoration(labelText: 'Confirm Password'),
                obscureText: true,
                validationMessages: (control) => {
                  ValidationMessage.mustMatch:
                      'Password confirmation must match',
                },
                onSubmitted: () => form.focus('rememberMe'),
                textInputAction: TextInputAction.next,
              ),
              const SizedBox(height: 24.0),
              RaisedButton(
                color: Colors.teal,
                padding: EdgeInsets.all(10.0),
                child: new Text(
                  "Focus to Birthday",
                  style: new TextStyle(
                    color: Colors.white,
                    fontSize: 16.0
                  ),
                ),
                onPressed: () {
                    form.control('dateTime').focus();
                }
              ),
              const SizedBox(height: 24.0),
              RaisedButton(
                color: Colors.teal,
                padding: EdgeInsets.all(10.0),
                child: new Text(
                  "Focus to Email",
                  style: new TextStyle(
                    color: Colors.white,
                    fontSize: 16.0
                  ),
                ),
                onPressed: () {
                    form.control('email').focus();
                }
              ),
            ],
          );
        },
      ),
    );
  }
}

/// Async validator in use emails example
const inUseEmails = ['johndoe@email.com', 'john@email.com'];

/// Async validator example that simulates a request to a server
/// to validate if the email of the user is unique.
Future<Map<String, dynamic>> _uniqueEmail(
    AbstractControl<dynamic> control) async {
  final error = {'unique': false};

  final emailAlreadyInUse = await Future.delayed(
    const Duration(seconds: 5), // delay to simulate a time consuming operation
    () => inUseEmails.contains(control.value.toString()),
  );

  if (emailAlreadyInUse) {
    control.markAsTouched();
    return error;
  }

  return null;
}
`

So basically I added two buttons (1 that focus on an email ReactiveTextField and 1 that focus on ReactiveDatePicker). Scroll on focus works in ReactiveTextField but not with the ReactiveDatePicker

**Comment by vasilich6107** [comment]
Hi @mammba01012018 
`ReactiveDatePicker` technically not the same as `ReactiveTextField`
It uses `InputDecorator` to mimic the look and feel.
But there actual `scroll into view` is implemented deeper in `Editable` widget(or something like this)
